addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -91,6 +99,58 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
+func (r *MirrorPeerReconciler) createS3(ctx context.Context) error {
+	var err error
+
+	noobaaOBCNamespace := &corev1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "namespace",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "openshift-dr-system",
+		},
+	}
+	err = r.SpokeClient.Get(ctx, types.NamespacedName{Name: "openshift-dr-system"}, noobaaOBCNamespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.Info("Could not find ODR OBC namespace, creating")
+			err = r.SpokeClient.Create(ctx, noobaaOBCNamespace)
+			if err != nil {
+				klog.Error(err, "Failed to create OBC namespace")
+				return err
+			}
+		} else {
+			klog.Error(err, "Failed to get OBC namespace")
+			return err
+		}
+	}
+
+	noobaaOBC := &obv1alpha1.ObjectBucketClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "odrbucket",
+			Namespace: "openshift-dr-system",
+		},
+		Spec: obv1alpha1.ObjectBucketClaimSpec{
+			StorageClassName: "openshift-storage.noobaa.io",
+		},
+	}
+	err = r.SpokeClient.Get(ctx, types.NamespacedName{Name: "odrbucket", Namespace: "openshift-dr-system"}, noobaaOBC)
```

Comment:
Try not to hardcode params that can change. What if ODR is installed in a different namespace?

Reply:
Actually, Gowtham and I confirmed with Shyam that the namespace in which the `odrbucket` is created will be `openshift-dr-system`.

Reply:
Ok, but if that changes we'll need to make another release just to change this. So, I want us to plan for that.

Reply:
you can try like this: https://github.com/red-hat-storage/odf-multicluster-orchestrator/blob/main/addons/token-exchange/blue_secret_controller.go#L64

Reply:
Agree with @umangachapagain although the namespace is as above, it would be better to have it configurable such that changes in the future can be accommodated.

Reply:
Assuming that the ramen-dr-cluster operator can create an `ODR_NAMESPACE` env var in the MC, this should suffice: https://github.com/red-hat-storage/odf-multicluster-orchestrator/pull/59/commits/ebed15d18cb772d602d0dcc56f222dfd8f8e4f21#diff-a6eb8372760bde67a512ec16d6e6c52c52b203e9ccc5d6cda10c78adf8f993bbR129